src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp

```diff
@@ -390,6 +422,10 @@ int XQSPIPS_Flasher::xclUpgradeFirmware(std::istream& binStream)
     binStream.seekg(0, binStream.beg);
 
     std::cout << "INFO: ***BOOT.BIN has " << total_size << " bytes" << std::endl;
+    if (total_size >= GOLDEN_BASE) {
+        std::cout << "[ERROR]: BOOT.BIN is too large. Flashing will overwrite golden." << std::endl;
+        exit(-EINVAL);
+    }
```

Comment:
Please clearly say that the BOOT.BIN should not bigger than 96MB.

BTW, I already let U30 platform developer know this 96MB change. They think it is okay. The typical size of BOOT.BIN is 96MB.

Reply:
changed.